utils.go

```diff
@@ -25,22 +25,68 @@ func queryParseDefault[V QueryType](err error, parser func() V, defaultValue ...
 	return parser()
 }
 
-func queryParseInt[V QueryType](q string, bitSize int, parser func(int64) V, defaultValue ...V) V {
-	result, err := strconv.ParseInt(q, 10, bitSize)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseInt[V GenericType](str string, bitSize int, parser func(int64) V, defaultValue ...V) V {
+	result, err := strconv.ParseInt(str, 10, bitSize)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
 }
 
-func queryParseUint[V QueryType](q string, bitSize int, parser func(uint64) V, defaultValue ...V) V {
-	result, err := strconv.ParseUint(q, 10, bitSize)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseUint[V GenericType](str string, bitSize int, parser func(uint64) V, defaultValue ...V) V {
+	result, err := strconv.ParseUint(str, 10, bitSize)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
 }
 
-func queryParseFloat[V QueryType](q string, bitSize int, parser func(float64) V, defaultValue ...V) V {
-	result, err := strconv.ParseFloat(q, bitSize)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseFloat[V GenericType](str string, bitSize int, parser func(float64) V, defaultValue ...V) V {
+	result, err := strconv.ParseFloat(str, bitSize)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
 }
 
-func queryParseBool[V QueryType](q string, parser func(bool) V, defaultValue ...V) V {
-	result, err := strconv.ParseBool(q)
-	return queryParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+func genericParseBool[V GenericType](str string, parser func(bool) V, defaultValue ...V) V {
+	result, err := strconv.ParseBool(str)
+	return genericParseDefault[V](err, func() V { return parser(result) }, defaultValue...)
+}
+
+func genericParseType[V GenericType](ctx *DefaultCtx, str string, v V, defaultValue ...V) V {
```

Comment:
maybe we should export this function

you can also add the converter function
```go
Convert[T any](value string, convertor func(string) (T, error), defaultValue ...T) (*T, error) { 
	converted, err := convertor(value)
	if err != nil {
		if len(defaultValue) > 0 {
			return &defaultValue[0], nil
		}

		return nil, fmt.Errorf("failed to convert: %w", err)
	}

	return &converted, nil
}
```

Reply:
I plan to add this function in the following PR when implementing the `GetGeneric` function.
Should I add the `GetGeneric` function to this PR, too?

Reply:
yes would be ok for me

Reply:
sorry i meant we can add it right here in this pull request to minimize the intermediate steps

you just need to change the description and headline of the pull request

Reply:
Sure. I'm gonna change the pull request headline and description.